### ISSUE TITLE
False positive Nextcloud dav sharing - 949110

### ISSUE THREAD

**Issue by Silur** [coreruleset/coreruleset#3085]
I have a nextcloud configured nginx server with crs set up and the nextcloud exclusions enabled by `setvsetvar:tx.crs_exclusions_nextcloud=1`

However, when I try to click the "share" icon on any file on my nextcloud server, the URL `https:/<nextcloud>/remote.php/dav/files/silur/<filename>` gets matched against an evaluation anomaly as follows:

``
2023/01/11 16:13:12 [error] 5004#5004: *3001 [client 84.17.53.222] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "80"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "REDACTED"] [uri "/remote.php/dav/files/silur/REDACTED"] [unique_id "REDACTED"] [ref ""], client: REDACTED, server: REDACTED, request: "PROPFIND /remote.php/dav/files/silur/REDACTED HTTP/2.0", host: "REDACTED", referrer: "https://REDACTED/apps/files/?dir=%2F&fileid=2"
``

Disabling the rule by Id works as a temporary solution but creates a large attack surface.

**Comment by azurit** [comment]
Hi @Silur, rule ID 949110 is only a blocking evaluation based on other matched rules. Can you post also other logs related to this request? Thank you.

**Comment by Silur** [comment]
```
bunker-bunker-1    | 2023/01/12 09:52:38 [warn] 57827#57827: *10311 ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?:get|post|head|options|connect|put|delete|trace|track|patch|propfind|propatch|mkcol|copy|move|lock|unlock)\s+(?:\/|\w)[^\s]*(?:\s+http\/\d|[\r\n])' against variable `REQUEST_BODY' (Value: `<?xml version="1.0"?>\x0a\x09\x09\x09<d:propfind  xmlns:d="DAV:"\x0a\x09\x09\x09\x09xmlns:oc="http:/ (947 characters omitted)' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-921-PROTOCOL-ATTACK.conf"] [line "33"] [id "921110"] [rev ""] [msg "HTTP Request Smuggling Attack"] [data "Matched Data: propfind  xmlns:d="dav:"\x0a found within REQUEST_BODY: <?xml version="1.0"?>\x0a\x09\x09\x09<d:propfind  xmlns:d="dav:"\x0a\x09\x09\x09\x09xmlns:oc="http://owncloud.org/ns"\x0a\x09\x09\x09\x09xmlns:nc="http://nextcloud.org/ns"\x0a\x09\x09\x09 (536 characters omitted)"] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272/220/33"] [hostname "10.20.30.3"] [uri "/remote.php/dav/files/silur//REDACTED"] [unique_id "167351715862.942049"] [ref "o28,25v1046,669t:urlDecodeUni,t:htmlEntityDecode,t:lowercase"], client: 89.37.173.32, server: REDACTED, request: "PROPFIND /remote.php/dav/files/silur//REDACTED HTTP/2.0", host: "REDACTED", referrer: "REDACTED"
bunker-bunker-1    | 2023/01/12 09:52:38 [error] 57827#57827: *10311 [client 89.37.173.32] ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/opt/bunkerweb/core/modsecurity/files/coreruleset/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "80"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.3.2"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "REDACTED"] [uri "/remote.php/dav/files/silur//REDACTED"] [unique_id "167351715862.942049"] [ref ""], client: REDACTED, server: REDACTED, request: "PROPFIND /remote.php/dav/files/silur//REDACTED HTTP/2.0", host: "REDACTED", referrer: "REDACTED"
```

**Comment by Silur** [comment]
I see now there is https://github.com/coreruleset/coreruleset/issues/3024 regarding the 921110 but it's not resolved (besides disabling the whole check on an endpoint)

**Comment by Silur** [comment]
```
SecRule REQUEST_URI "@beginsWith /remote.php/dav" \ 
"id:1000,\
phase:1,\
pass,\
nolog,\
ctl:ruleRemoveById=921110
ctl:ruleRemoveById=949110"
```

Addig the rule above to the crs config also didn't work, the rule is catched as a warning then caught as an error with 949110.

**Comment by azurit** [comment]
This is not related to #3024 and seems to be already covered by rule `9508106` which is part of [Nextcloud Rule Exclusions Plugin](https://github.com/coreruleset/nextcloud-rule-exclusions-plugin).

**Comment by Silur** [comment]
> This is not related to #3024 and seems to be already covered by rule `9508106` which is part of [Nextcloud Rule Exclusions Plugin](https://github.com/coreruleset/nextcloud-rule-exclusions-plugin).

Indeed that one is a different ID sorry :)
Nevertheless, disabling  921110 on the endpoint did work as a temporary solution

**Comment by azurit** [comment]
Just install a new plugin instead of RE package.

**Comment by jessebot** [comment]
Sorry to bother, but what does RE mean?

**Comment by azurit** [comment]
@jessebot Rule Exclusion package. This plugin is some kind of a new version of old RE package.

**Comment by jessebot** [comment]
ah, thanks!

**Comment by EsadCetiner** [comment]
@azurit any reason why this is re-opened? This should be fixed in the plugin https://github.com/coreruleset/nextcloud-rule-exclusions-plugin/blob/b21c032d487e8d998fb5a49dcc57066acee11396/plugins/nextcloud-rule-exclusions-before.conf#L170
